<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="icon" href="images/rnb_favicon.ico">

    <title>DecompProtocol</title>

    <!-- Bootstrap Core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="css/small-business.css" rel="stylesheet">

</head>

<body>

    <!-- Navigation -->
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container">
            <!-- Brand and toggle get grouped for better mobile display -->
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                    <span class="sr-only">Toggle navigation</span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="index.html">
                    <b>DecompProtocol</b>
                </a>
            </div>
            <!-- Collect the nav links, forms, and other content for toggling -->
            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav">
                    <li>
                        <a href="index.html">About</a>
                    </li>
					<li class="active">
						<a href="protocol.html">Protocol</a>
					</li>
                    <li>
                        <a href="resources.html">Resources</a>
                    </li>
               </ul>
            </div>
            <!-- /.navbar-collapse -->
        </div>
        <!-- /.container -->
    </nav>
<div class="container">
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This protocol aims at guiding reasearcher how to employ deconvolution
 of methylomes obtained from complex tissues. It will start with data 
retrieval from a public resource, but is equally applicable to in-house 
generated data. We will furthermore focus on the Illumina BeadChip 
series as a data source, although the protocol is also compatible with 
bisulfite sequencing, or any technology yielding single base pair 
resolution. Deconvolution here refers to creating two matrices 
(proportion matrix A and methylation pattern matrix T) from a single 
matrix of input DNA methylation data (dimension CpGs x samples). 
Non-negative matrix factorization is employed for this task, and we will
 discuss some of the advantages and caveats of the methods.</p>
</div>
<div id="installation-duration-up-to-2-h" class="section level1">
<h1>Installation (duration up to 2 h)</h1>
<ol style="list-style-type: decimal">
<li><p>If R is not yet installed, follow the instructions at <a href="https://cran.r-project.org/" class="uri">https://cran.r-project.org/</a>.
 Create a working directory on a filesystem partition with sufficient 
storage capacity. Throughout the analysis, ~20-30 Gb of free storage 
will be required. Be sure that all the files are downloaded into this 
working directory and that the code is executed within the directory.</p></li>
<li><p>To execute the protocol described here, several R-packages (<em>RnBeads</em>, <em>DecompPipeline</em>, <em>MeDeCom</em>, and <em>FactorViz</em>)
 are required. First, you should have a recent R version installed and 
all your packages updated. The installation of the software is appliable
 for most <em>Linux distributions</em> directly through R, while for <em>MacOS</em> the binary release of <em>MeDeCom</em> <a href="https://github.com/lutsik/MeDeCom/releases/download/v0.3.0/MeDeCom_0.3.0.tgz" class="uri">https://github.com/lutsik/MeDeCom/releases/download/v0.3.0/MeDeCom_0.3.0.tgz</a> should be used. The protocol is also available as a Docker container <a href="https://hub.docker.com/r/mscherer/medecom" class="uri">https://hub.docker.com/r/mscherer/medecom</a>,
 which is the only option for Windows operating systems. For the 
remainder of this protocol, we assume a common Linux distribution as the
 operating system.</p></li>
</ol>
<pre class="r"><code class="hljs">install.packages(c(<span class="hljs-string">"devtools"</span>,<span class="hljs-string">"BiocManager"</span>))
BiocManager::install(c(<span class="hljs-string">"RnBeads"</span>,<span class="hljs-string">"RnBeads.gh19"</span>,<span class="hljs-string">"RnBeads.mm10"</span>,<span class="hljs-string">"RnBeads.hg38"</span>),dependencies=<span class="hljs-literal">TRUE</span>)
devtools::install_github(c(<span class="hljs-string">"lutsik/MeDeCom"</span>,<span class="hljs-string">"CompEpigen/DecompPipeline"</span>,<span class="hljs-string">"CompEpigen/FactorViz"</span>))
<span class="hljs-keyword">library</span>(DecompPipeline)</code></pre>
</div>
<div id="protocol" class="section level1">
<h1>Protocol</h1>
<div id="data-retrival" class="section level2">
<h2>Data Retrival</h2>
<div id="obtaining-data-from-a-public-resource-duration-5h" class="section level3">
<h3>Obtaining data from a public resource (duration ~5h)</h3>
<ol start="3" style="list-style-type: decimal">
<li>Use the Genomic Data Commons (GDC) data download tool (<a href="https://gdc.cancer.gov/access-data/gdc-data-transfer-tool" class="uri">https://gdc.cancer.gov/access-data/gdc-data-transfer-tool</a>) to download the IDAT files with the TCGA-LUAD manifest file (<a href="http://epigenomics.dkfz.de/DecompProtocol/data/gdc_manifest.2019-01-23.txt" class="uri">http://epigenomics.dkfz.de/DecompProtocol/data/gdc_manifest.2019-01-23.txt</a>) and its associated metadata by running the download tool in a terminal session initiated in the working directory:</li>
</ol>
<pre class="bash"><code class="hljs">gdc-client download –m gdc_manifest.2019-01-23.txt</code></pre>
<p>Store the resulting files in a new directory <em>idat</em>.</p>
<ol start="4" style="list-style-type: decimal">
<li>Retrieve the clinical metadata from <a href="https://portal.gdc.cancer.gov/projects/TCGA-LUAD" class="uri">https://portal.gdc.cancer.gov/projects/TCGA-LUAD</a> by clicking on the “Clinical” button and unpacking the downloaded <em>.tar.gz</em>
 file into a new subdirectory annotation within your working directory. 
The remaining parsing is performed through an R session initiated in the
 same directory:</li>
</ol>
<pre class="r"><code class="hljs">clinical.data &lt;- read.table(<span class="hljs-string">"annotation/clinical.tsv"</span>,sep=<span class="hljs-string">"\t"</span>,header=<span class="hljs-literal">T</span>)
idat.files &lt;- list.files(<span class="hljs-string">"idat"</span>,full.names = <span class="hljs-literal">T</span>)
meta.files &lt;- list.files(idat.files[<span class="hljs-number">1</span>],full.names = <span class="hljs-literal">T</span>)
untar(meta.files[<span class="hljs-number">3</span>],exdir = idat.files[<span class="hljs-number">1</span>])
meta.files &lt;- untar(meta.files[<span class="hljs-number">3</span>],list=<span class="hljs-literal">T</span>)
meta.info &lt;- read.table(file.path(idat.files[<span class="hljs-number">1</span>],meta.files[<span class="hljs-number">5</span>]),sep=<span class="hljs-string">"\t"</span>,header=<span class="hljs-literal">T</span>)
meta.info &lt;- meta.info[match(unique(meta.info$Comment..TCGA.Barcode.),meta.info$Comment..TCGA.Barcode.),]
match.meta.clin &lt;- match(clinical.data$submitter_id,substr(meta.info$Comment..TCGA.Barcode.,<span class="hljs-number">1</span>,<span class="hljs-number">12</span>))
anno.frame &lt;- na.omit(data.frame(clinical.data,meta.info[match.meta.clin,]))
anno.frame$barcode &lt;- unlist(lapply(lapply(as.character(anno.frame$Array.Data.File),<span class="hljs-keyword">function</span>(x)strsplit(x,<span class="hljs-string">"_"</span>)),<span class="hljs-keyword">function</span>(x)paste(x[[<span class="hljs-number">1</span>]][<span class="hljs-number">1</span>],x[[<span class="hljs-number">1</span>]][<span class="hljs-number">2</span>],sep=<span class="hljs-string">"_"</span>)))
anno.frame$Sentrix_ID &lt;- unlist(lapply(lapply(as.character(anno.frame$Array.Data.File),<span class="hljs-keyword">function</span>(x)strsplit(x,<span class="hljs-string">"_"</span>)),<span class="hljs-keyword">function</span>(x)paste(x[[<span class="hljs-number">1</span>]][<span class="hljs-number">1</span>])))
anno.frame$Sentrix_Position &lt;- unlist(lapply(lapply(as.character(anno.frame$Array.Data.File),<span class="hljs-keyword">function</span>(x)strsplit(x,<span class="hljs-string">"_"</span>)),<span class="hljs-keyword">function</span>(x)paste(x[[<span class="hljs-number">1</span>]][<span class="hljs-number">2</span>])))
anno.frame$healthy_cancer &lt;- ifelse(grepl(<span class="hljs-string">"11A"</span>,anno.frame$Comment..TCGA.Barcode.),<span class="hljs-string">"healthy"</span>,<span class="hljs-string">"cancer"</span>)
write.table(anno.frame,<span class="hljs-string">"annotation/sample_annotation.tsv"</span>,quote=<span class="hljs-literal">F</span>,row.names = <span class="hljs-literal">F</span>,sep=<span class="hljs-string">"\t"</span>)
anno.frame &lt;- read.table(<span class="hljs-string">"annotation/sample_annotation.tsv"</span>,quote=<span class="hljs-literal">F</span>,row.names = <span class="hljs-literal">F</span>,sep=<span class="hljs-string">"\t"</span>)</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>Copy the IDAT files into a single directory idat for downstream analysis.</li>
</ol>
<pre class="r"><code class="hljs"><span class="hljs-comment">#' write idat files to parent directory</span>
lapply(idat.files,<span class="hljs-keyword">function</span>(x){
  is.idat &lt;- list.files(x,pattern = <span class="hljs-string">".idat"</span>,full.names = <span class="hljs-literal">T</span>)
  file.copy(is.idat,<span class="hljs-string">"idat/"</span>)
  unlink(x,recursive = <span class="hljs-literal">T</span>)
})</code></pre>
</div>
</div>
<div id="data-processing" class="section level2">
<h2>Data Processing</h2>
<div id="data-import-2h" class="section level3">
<h3>Data import (~2h)</h3>
<ol start="6" style="list-style-type: decimal">
<li><em>RnBeads</em> converts the files into a data object and performs basic quality control. Analysis options have to be specified for <em>RnBeads</em>,
 either through an XML file, or through the command line. Deactivate the
 preprocessing, exploratory, covariate inference, export and 
differential methylation modules, such that <em>RnBeads</em> only performs data import and quality control.</li>
</ol>
<pre class="r"><code class="hljs">suppressPackageStartupMessages(<span class="hljs-keyword">library</span>(RnBeads))
rnb.options(
  assembly=<span class="hljs-string">"hg19"</span>,
  identifiers.column=<span class="hljs-string">"submitter_id"</span>,
  import=<span class="hljs-literal">T</span>,
  import.default.data.type=<span class="hljs-string">"idat.dir"</span>,
  import.table.separator=<span class="hljs-string">"\t"</span>,
  import.sex.prediction=<span class="hljs-literal">T</span>,
  qc=<span class="hljs-literal">T</span>,
  preprocessing=<span class="hljs-literal">F</span>,
  exploratory=<span class="hljs-literal">F</span>,
  inference=<span class="hljs-literal">F</span>,
  differential=<span class="hljs-literal">F</span>,
  export.to.bed=<span class="hljs-literal">F</span>,
  export.to.trackhub=<span class="hljs-literal">NULL</span>,
  export.to.csv=<span class="hljs-literal">F</span>
)</code></pre>
<ol start="7" style="list-style-type: decimal">
<li>Specify the input to <em>RnBeads</em>: the sample annotation sheet 
created at the data retrieval step, the directory in which the IDAT 
files are stored and a directory to which the HTML report is to be 
saved. Additionally, specify a temporary directory and start the RnBeads
 analysis.</li>
</ol>
<pre class="r"><code class="hljs">sample.anno &lt;- <span class="hljs-string">"annotation/sample_annotation.tsv"</span>
idat.folder &lt;- <span class="hljs-string">"idat/"</span>
dir.report &lt;- paste0(<span class="hljs-string">"report"</span>,Sys.Date(),<span class="hljs-string">"/"</span>)
temp.dir &lt;- <span class="hljs-string">"/tmp"</span>
options(fftempdir=temp.dir)
rnb.set &lt;- rnb.run.analysis(dir.reports = dir.report, sample.sheet = sample.anno, data.dir = idat.folder)</code></pre>
<p><strong>PAUSE POINT</strong> We provide an example report containing the processed RnBSet object for further analysis. It can be obtained from <a href="http://epigenomics.dkfz.de/downloads/DecompProtocol/RnBeads_Report_TCGA_LUAD/index.html" class="uri">http://epigenomics.dkfz.de/downloads/DecompProtocol/RnBeads_Report_TCGA_LUAD/index.html</a> or directly loaded via:</p>
<pre class="r"><code class="hljs">download.file(<span class="hljs-string">"http://epigenomics.dkfz.de/downloads/DecompProtocol/rnbSet_unnormalized.zip"</span>, destfile=paste0(tempdir(),<span class="hljs-string">"/RnBSet"</span>))
rnb.set &lt;- load.rnb.set(paste0(tempdir(),<span class="hljs-string">"/RnBSet"</span>))</code></pre>
<ol start="8" style="list-style-type: decimal">
<li><em>RnBeads</em> creates an interactive HTML report, specifying the 
steps performed and the associated results. Data was of good quality 
such that it can be used for further analysis.</li>
</ol>
<p><strong>PAUSE POINT</strong> For reference, we provide a complete RnBeads report on the supplementary website <a href="http://epigenomics.dkfz.de/downloads/DecompProtocol/RnBeads_Report_TCGA_LUAD/" class="uri">http://epigenomics.dkfz.de/downloads/DecompProtocol/RnBeads_Report_TCGA_LUAD/</a>.</p>
</div>
<div id="preprocessing-and-filtering-22h" class="section level3">
<h3>Preprocessing and Filtering (22h)</h3>
<ol start="9" style="list-style-type: decimal">
<li>For further data preparation and analysis steps use the <em>DecompPipeline</em>
 package. Processing options are provided through individual function 
parameters. Follow a stringent filtering strategy: (i) Filter CpGs 
covered by less than 3 beads, and probes that are in the 0.001 and 0.999
 overall intensity quantiles (low and high intensity outliers). (ii) 
Remove all probes containing missing values in any of the samples. (iii)
 Discard sites outside of CpG context, overlapping annotated SNPs, 
located on the sex chromosomes and potentially cross-reactive probes. 
Finally, apply BMIQ normalization to account for the bias introduced by 
the two Infinium probe designs.</li>
</ol>
<p><strong>PAUSE POINT</strong> We provide a list of selected CpGs as an intermediate result <a href="http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_complete_filtering.csv" class="uri">http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_complete_filtering.csv</a>.</p>
<pre class="r"><code class="hljs">rem.sites &lt;- rep(<span class="hljs-literal">TRUE</span>,nrow(meth(rnb.set)))
sel.sites &lt;-
read.csv(<span class="hljs-string">"http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_co
mplete_filtering.csv"</span>)
rem.sites[row.names(annotation(rnb.set))%<span class="hljs-keyword">in</span>%as.character(sel.sites[,<span class="hljs-number">1</span>])] &lt;-
<span class="hljs-literal">FALSE</span>
rnb.set &lt;- remove.sites(rnb.set,rem.sites)
data.prep &lt;- list(rnb.set.filtered=rnb.set)</code></pre>
<ol start="10" style="list-style-type: decimal">
<li>Adjustment for potential confounding factors is crucial in 
epigenomic studies and the influences of, for instance, donor sex, age, 
and genotype on the DNA methylation pattern are well-studied 28,29 . Use
 Independent Component Analysis (ICA, see Materials) to account for DNA 
methylation differences that are due to sex, age, race, and ethnicity.</li>
</ol>
<pre class="r"><code class="hljs">suppressPackageStartupMessages(<span class="hljs-keyword">library</span>(DecompPipeline))
data.prep &lt;- prepare_data(RNB_SET = rnb.set,
                          analysis.name = <span class="hljs-string">"TCGA_LUAD"</span>,
                          NORMALIZATION = <span class="hljs-string">"bmiq"</span>,
                          FILTER_BEADS = <span class="hljs-literal">T</span>,
                          MIN_N_BEADS = <span class="hljs-number">3</span>,
                          FILTER_INTENSITY = <span class="hljs-literal">T</span>,
                          MIN_INT_QUANT = <span class="hljs-number">0.001</span>,
                          MAX_INT_QUANT = <span class="hljs-number">0.999</span>,
                          FILTER_NA = <span class="hljs-literal">T</span>,
                          FILTER_CONTEXT = <span class="hljs-literal">T</span>,
                          FILTER_SNP = <span class="hljs-literal">T</span>,
                          FILTER_SOMATIC = <span class="hljs-literal">T</span>,
                          FILTER_CROSS_REACTIVE = <span class="hljs-literal">T</span>,
                          execute.lump=<span class="hljs-literal">T</span>,
                          remove.ICA=<span class="hljs-literal">T</span>,
                          conf.fact.ICA=c(<span class="hljs-string">"age_at_diagnosis"</span>,<span class="hljs-string">"race"</span>,<span class="hljs-string">"gender"</span>,<span class="hljs-string">"ethnicity"</span>),
                          ica.setting=c(<span class="hljs-string">"alpha.fact"</span>=<span class="hljs-number">1e-5</span>,<span class="hljs-string">"save.report"</span>=<span class="hljs-literal">T</span>,<span class="hljs-string">"nmin"</span>=<span class="hljs-number">30</span>,<span class="hljs-string">"nmax"</span>=<span class="hljs-number">50</span>,<span class="hljs-string">"ncores"</span>=<span class="hljs-number">10</span>))</code></pre>
<p><strong>BREAKPOINT</strong> We provide a list of CpGs that are selected for the downstream analysis at <a href="http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_complete_filtering.csv" class="uri">http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_complete_filtering.csv</a>. You can directly select those sites from you RnBSet object.</p>
<pre class="breakpoint_2"><code class="hljs">rem.sites &lt;- rep(TRUE,nrow(meth(rnb.set)))
sel.sites &lt;- read.csv("http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_complete_filtering.csv")
rem.sites[row.names(annotation(rnb.set))%in%as.character(sel.sites[,1])] &lt;- FALSE
rnb.set &lt;- remove.sites(rnb.set,rem.sites)
data.prep &lt;- list(rnb.set.filtered=rnb.set)</code></pre>
</div>
<div id="selecting-informative-features-cpgs" class="section level3">
<h3>Selecting informative features (CpGs)</h3>
<ol start="11" style="list-style-type: decimal">
<li>Select a subset of sites to be used for deconvolution. DecompPipeline provides different options (see documentation of <em>prepare_CG_subsets</em>) through the <em>prepare_CG_subsets</em> function. Select the 5,000 most highly variable CpGs across the samples.</li>
</ol>
<pre class="r"><code class="hljs">cg_subset &lt;- prepare_CG_subsets(rnb.set=data.prep$rnb.set.filtered,
                                MARKER_SELECTION = <span class="hljs-string">"var"</span>,
                                N_MARKERS = <span class="hljs-number">5000</span>)
names(cg_subset)</code></pre>
</div>
</div>
<div id="methylome-deconvolution" class="section level2">
<h2>Methylome Deconvolution</h2>
<div id="performing-deconvolution" class="section level3">
<h3>Performing Deconvolution</h3>
<ol start="12" style="list-style-type: decimal">
<li>Perform the deconvolution experiment. Use MeDeCom with a grid of 
values for the number of components (K) ranging from 2 to 15, which 
covers homogeneous to heterogeneous samples. Also, specify a grid for 
the regularization parameter (λ) from strong (0.01) to no regularization
 (0). We here focus on <em>MeDeCom</em> as the Deconvolution tool, although <em>DecompPipeline</em> also supports <em>RefFreeCellMix</em> and <em>EDec</em>.</li>
</ol>
<pre class="r"><code class="hljs">md.res &lt;- start_medecom_analysis(
  rnb.set=data.prep$rnb.set.filtered,
  cg_groups = cg_subset,
  Ks=<span class="hljs-number">2</span>:<span class="hljs-number">15</span>,
  LAMBDA_GRID = c(<span class="hljs-number">0</span>,<span class="hljs-number">10</span>^-(<span class="hljs-number">2</span>:<span class="hljs-number">5</span>)),
  factorviz.outputs = <span class="hljs-literal">T</span>,
  analysis.name = <span class="hljs-string">"TCGA_LUAD"</span>,
  cores = <span class="hljs-number">15</span>
)</code></pre>
<p><strong>PAUSE POINT</strong> The final MeDeCom result is stored in a 
format that can be directly imported with FactorViz. We provide the 
final result for exploration at <a href="http://epigenomics.dkfz.de/downloads/DecompProtocol/FactorViz_outputs.tar.gz" class="uri">http://epigenomics.dkfz.de/downloads/DecompProtocol/FactorViz_outputs.tar.gz</a>.</p>
<pre class="r"><code class="hljs">download.file(<span class="hljs-string">"http://epigenomics.dkfz.de/downloads/DecompProtocol/FactorViz_outputs.tar.gz"</span>,destfile=paste0(tempdir(),<span class="hljs-string">"/FactorViz_outputs.tar.gz"</span>))
untar(paste0(tempdir(),<span class="hljs-string">"/FactorViz_outputs.tar.gz"</span>))</code></pre>
</div>
</div>
<div id="downstream-analysis" class="section level2">
<h2>Downstream analysis</h2>
<ol start="13" style="list-style-type: decimal">
<li>Start the <em>FactorViz</em> application to visualize and interactively explore the deconvolution results.</li>
</ol>
<pre class="r"><code class="hljs">suppressPackageStartupMessages(<span class="hljs-keyword">library</span>(FactorViz))
startFactorViz(file.path(getwd(),<span class="hljs-string">"TCGA_LUAD"</span>,<span class="hljs-string">"FactorViz_outputs"</span>))</code></pre>
<ol start="14" style="list-style-type: decimal">
<li><p>Determine the number of LMCs (K) and the regularization parameter
 (λ) based on the cross-validation error. First, go to panel “K 
selection” to plot cross-validation error for the range of Ks specified 
earlier. We recommend selecting K values as a trade-off between too high
 complexity, i.e.&nbsp;fitting the noise in the data (higher K values) 
and insufficient degrees of freedom (lower K values). This typically 
corresponds to the saddle point where cross-validation error starts to 
level out.</p></li>
<li><p>For a fixed K, select a value for the regularization parameter 
(λ) by proceeding to panel “Lambda selection”. An optimal value for λ 
will often correspond to a local minimum of the cross-validation error. 
On the other hand, noticeable changes in other statistics, such as 
objective value or root mean squared error (RMSE), can point at a 
different λ value.</p></li>
<li><p>In the “Proportions” panel of FactorViz, visualize LMC 
proportions using heatmaps. Proportion heatmaps can be visually 
annotated with available qualitative and quantitative traits (see 
dropdown “Color samples by”). Further visualization options include 
stacked barplot and lineplots.</p></li>
<li><p>In the “Meta-analysis” panel, associate LMC proportions with 
quantitative and qualitative traits using correlation- and t-tests. 
Further sample annotations, such as mutational load (<a href="https://www.cbioportal.org/study/clinicalData?id=luad_tcga_pan_can_atlas_2018" class="uri">https://www.cbioportal.org/study/clinicalData?id=luad_tcga_pan_can_atlas_2018</a>) and tumor purity scores can be obtained from public repositories and related publications (<a href="https://static-content.springer.com/esm/art%3A10.1038%2Fncomms3612/MediaObjects/41467_2013_BFncomms3612_MOESM489_ESM.xlsx">https://static-content.springer.com/esm/art%3A10.1038%2Fncomms3612/MediaObjects/41467_2013_BFncomms3612_MOESM489_ESM.xlsx</a>). In addition, scripts to conduct such analysis are available on the Supplementary Website (<a href="http://epigenomics.dkfz.de/DecompProtocol/" class="uri">http://epigenomics.dkfz.de/DecompProtocol/</a>).</p></li>
<li><p>Explore the LMCs in the panel “LMCs”. Several visualization 
options are available such as Multidimensional Scaling and histograms. 
Reference profiles can be used for joint visualization in case those are
 available.</p></li>
<li><p>Determine DNA methylation sites that are specifically hypo- and 
hypermethylated in an LMC by comparing the methylation values in the LMC
 matrix for each LMC to the median of the remaining LMCs in the 
“Meta-analysis” panel. LMC-specific sites can be used for either a 
gene-centric Gene Ontology analysis (select ”Enrichments” and “GO 
Enrichments” in dropdown ”Analysis” and “Output type”) or for 
region-based enrichment analysis with the LOLA package (select “LOLA 
Enrichments” in dropdown ”Output type”). The differential sites can be 
exported for further analysis, and the resulting plots can be stored by 
clicking on the “PDF” button.</p></li>
</ol>
<p><strong>PAUSE POINT</strong> Additional built-in options for visualization of LMCs and proportions are available via the <em>MeDeCom</em> functions <code>plotLMCs</code> and <code>plotProportions</code> in R. Finally, the LMC and proportions matrix can be extracted from the MeDeComSet object, and the genomic annotation of CpGs (<code>ann.C</code>) can be obtained from the <em>FactorViz</em> output for a custom downstream analysis:</p>
<pre class="r"><code class="hljs">LMCs &lt;- getLMCs(medecom.set, K=<span class="hljs-number">7</span>, lambda=<span class="hljs-number">0.001</span>)
LMC.proportions &lt;- getProportions(medecom.set, K=<span class="hljs-number">7</span>, lambda=<span class="hljs-number">0.001</span>)
load(<span class="hljs-string">"FactorViz_outputs/ann_C.RData"</span>)
head(ann.C)</code></pre>
</div>
</div>    <footer class="py-5 bg-inverse">
        <div class="container">           <p class="m-0 text-center text-white">Copyright &copy; <a href="mailto:mscherer@mpi-inf.mpg.de">Support</a> 2019 </p>
        </div>
    </footer>

    <!-- jQuery -->
    <script src="js/jquery.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="js/bootstrap.min.js"></script>

</body>

</html>
