<!DOCTYPE html>
<html lang="en">

<head>

    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">
    <link rel="icon" href="images/rnb_favicon.ico">

    <title>DecompProtocol</title>

    <!-- Bootstrap Core CSS -->
    <link href="css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom CSS -->
    <link href="css/small-business.css" rel="stylesheet">

</head>

<body>

    <!-- Navigation -->
    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
        <div class="container">
            <!-- Brand and toggle get grouped for better mobile display -->
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                    <span class="sr-only">Toggle navigation</span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                <a class="navbar-brand" href="index.html">
                    <b>DecompProtocol</b>
                </a>
            </div>
            <!-- Collect the nav links, forms, and other content for toggling -->
            <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                <ul class="nav navbar-nav">
                    <li>
                        <a href="index.html">About</a>
                    </li>
			<li class="dropdown-active">
				<a class="dropdown-toggle" data-toggle="dropdown" href="#">Protocol
				<span class="caret"></span></a>
				<ul class="dropdown-menu">
					<li><a href="protocol.html">Protocol for 450k data (MeDeCom)</a></li>
					<li><a href="protocol_RefFreeEWAS.html">Protocol for 450k data (RefFreeEWAS)</a></li>
					<li><a href="protocol_RRBS.html">Protocol for RRBS data (MeDeCom)</a></li>
				</ul>
			</li>
			</li>
                    <li>
                        <a href="resources.html">Resources</a>
                    </li>
               </ul>
            </div>
            <!-- /.navbar-collapse -->
        </div>
        <!-- /.container -->
    </nav>

    <!-- Page Content -->
    <div class="container">

        <!-- Page Heading/Breadcrumbs -->
        <h1 class="mt-4 mb-3">Supplementary Resources for Reference-Free Deconvolution of complex DNA methylation data</h1>
	
		<a name="tableDes"></a>
		<h2>Input data</h2>
		<p>The table below describes the input data used for the reference-free deconvolution protocol and how it can be obtained.</p>
        <div class="row">
           <div class="col-md-12">
               <table class="table table-hover">
                 <thead>
                   <tr>
                     <th>Name</th>
                     <th>URL</th>
                     <th>Description</th>
                  </tr>
                 </thead>
                 <tbody>
				   <tr>
						<td>GDC data download tool</td>
						<td><a href="https://gdc.cancer.gov/access-data/gdc-data-transfer-tool">https://gdc.cancer.gov/access-data/gdc-data-transfer-tool</a></td>
						<td>On the website, select the version most appropriate for your computational infrastructure and follow the instructions to install the tool.</td>
				   </tr>
                   <tr>
					 <td>Manifest file for TCGA-LUAD</td>
					 <td><a href="https://portal.gdc.cancer.gov/legacy-archive/search/f">https://portal.gdc.cancer.gov/legacy-archive/search/f</a></td>
					 <td>Select "TCGA" in the category "Cancer program" on the left, and the "Disease type" as "lung adenocarcinoma". Then, go to files on the top of the left hand side and select "Methylation array" under "Experimental strategy". Last, select "Raw intensities" under "Data type" and "Illumina Human Methylation 450" under "Platform" and click on "Download Manifest".</td>	
				  </tr>
				  <tr>
                     <td>Clinical metadata</td>
 					 <td><a href="https://portal.gdc.cancer.gov/projects/TCGA-LUAD">
https://portal.gdc.cancer.gov/projects/TCGA-LUAD
					</a></td>
                     <td>Click on the "Clinical" button and select "TSV" as the download option. Then, unpack the downloaded <i>.tar.gz</i> file into a new subdirectory
<i>annotation</i> within your working directory.</td>
                   </tr>
                   <tr>
                     <td>Mutational data</td>
                     <td><a href="https://www.cbioportal.org/study/clinicalData?id=luad_tcga_pan_can_atlas_2018">
https://www.cbioportal.org/study/clinicalData?id=luad_tcga_pan_can_atlas_2018
</a></td>
                     <td>Click on the download icon, which shows "Download clinical data for the selected cases" by dragging the mouse cursor over it and storing the resulting file <i>luad_tcga_pan_can_atlas_2018_clinical_data.tsv</i>. The file will be required as input in <a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/compare_with_mutational_data.R">this</a> plotting script.</td>
                   </tr>
					<tr>
                     <td>Tumor purity scores</td>
                     <td><a href="
https://static-
content.springer.com/esm/art%3A10.1038%2Fncomms3612/MediaObjects/41467_2013_BFncomms3
612_MOESM489_ESM.xlsx
">
https://static-
content.springer.com/esm/art%3A10.1038%2Fncomms3612/MediaObjects/41467_2013_BFncomms3
612_MOESM489_ESM.xlsx
</a></td>
                     <td>Download the Excel file and browse to the sheet "RNASeqV2". From this sheet, select the "lung adenocarcinoma" cases from the column "platform" and store the resulting sheet a tabular file named "LUAD_ESTIMATE_RNAseqV2.tab" in the <i>annotation</i> subdirectory. The file is needed for the <a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/trait_association.R">trait association script</a>.</td>
                   </tr>
				  <tr>
					<td>Stemness indices</td>
					<td><a href="https://ars.els-cdn.com/content/image/1-s2.0-S0092867418303581-mmc1.xlsx">https://ars.els-cdn.com/content/image/1-s2.0-S0092867418303581-mmc1.xlsx</a></td>
					<td>Download the excel file and browse to the sheet "StemnessScores_DNAmeth". Store this sheet as a new file "stemness_index.csv" in the directory "annotation". This file will be required to correlate LMC proportions with cancer stemness indices using <a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/correlation_with_stemness.R">this</a> script.</td>
				  </tr>
                 </tbody>
               </table>
           </div>
        </div>
		
		<a name="tableDes"></a>
		<h2>List of Resources and Intermediate Results</h2>
		<p>The table below describes the supplementary resources that are available for the reference-free deconvolution protocol.</p>
        <div class="row">
           <div class="col-md-12">               
               <table class="table table-hover">
                 <thead>
                   <tr>
                     <th>Name</th>
                     <th>URL</th>
                     <th>Description</th>
                  </tr>
                 </thead>
                 <tbody>
                   <tr>
					 <td>Manifest file for TCGA-LUAD</td>
					 <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/gdc_manifest.2019-01-23.txt">https://github.com/CompEpigen/Decomp_web/blob/master/data/gdc_manifest.2019-01-23.txt</a></td>
					 <td>The manifest file for the TCGA LUAD dataset. It can be used to download the IDAT files and associated metadata from TCGA using the <a href="https://gdc.cancer.gov/access-data/gdc-data-transfer-tool">GDC data transfer tool</a>.	
				  </tr>
				  <tr>
                     <td><i>RnBeads</i> report</td>
                     <td><a href="
http://epigenomics.dkfz.de/downloads/DecompProtocol/RnBeads_Report_TCGA_LUAD/
">
http://epigenomics.dkfz.de/downloads/DecompProtocol/RnBeads_Report_TCGA_LUAD/
</a></td>
                     <td>The <i>RnBeads</i> report generated from the lung adenocarcinoma dataset from TCGA (TCGA-LUAD). The protocol only requires the Import and Quality Control modules to be executed, but we provide a complete execution of the <i>RnBeads</i> pipeline including exploratory analysis.</td>
                   </tr>
                   <tr>
                     <td><i>RnBSet</i></td>
                     <td><a href="http://epigenomics.dkfz.de/downloads/DecompProtocol/rnbSet_unnormalized.zip">
http://epigenomics.dkfz.de/downloads/DecompProtocol/rnbSet_unnormalized.zip
</a></td>
                     <td>An processed <i>RnBSet</i> object comprising sample metadata, DNA methylation data, and CpG annotations. The dataset has not been subject to preprocessing and normalization.</td>
                   </tr>
					<tr>
                     <td>CpGs passing quality filtering</td>
                     <td><a href="
http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_quality_filtering.csv
">
http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_quality_filtering.csv
</a></td>
                     <td>A list of CpG identifiers from the EPIC array, which pass the stringent quality criteria for this particular dataset. The list comes as a comma-separated values (CSV) file.</td>
                   </tr>
					<tr>
                     <td>CpGs passing context filtering</td>
                     <td><a href="
http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_context_filtering.csv
">
http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_context_filtering.csv
</a></td>
                     <td>A list of CpG identifiers from the EPIC array, which pass the context filtering steps. In this step, sites annotated to single nucleotide polymorphisms, or to the sex chromosomes are removed.</td>
                   </tr>
					<tr>
                     <td>CpGs passing complete filtering</td>
                     <td><a href="
http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_complete_filtering.csv
">
http://epigenomics.dkfz.de/downloads/DecompProtocol/sites_passing_complete_filtering.csv
</a></td>
                     <td>A list of CpG identifiers from the EPIC array, which pass all filtering steps employed in <i>DecompPipeline</i>. This is an extension of the list above, which also removes sites that have been reported to be <a href="https://doi.org/10.4161/epi.23470">cross-reactive</a>.</td>
                   </tr>
					<tr>
                     <td><i>FactorViz</i> output</td>
                     <td><a href="
http://epigenomics.dkfz.de/downloads/DecompProtocol/FactorViz_outputs.tar.gz
">
http://epigenomics.dkfz.de/downloads/DecompProtocol/FactorViz_outputs.tar.gz
</a></td>
                     <td>This folder comprises the results of the deconvolution experiment of the TCGA LUAD dataset, which can be directly (after extraction) imported through <i>FactorViz</i>. The folder contains the <i>MeDeComSet</i>, genomic annotation of the CpGs, and sample metadata.</td>
                   </tr>
                 </tbody>
               </table>
           </div>
        </div>
        <!-- /.row -->

	<h2>Plotting scripts</h2>
		<p>The table below describes scripts that have been used to generate the plots shown within the protocol. All scripts are expected to be invoked in the working directory created during the execution of the protocol.</p>
        <div class="row">
           <div class="col-md-12">               
               <table class="table table-hover">
                 <thead>
                   <tr>
                     <th>Name</th>
                     <th>URL</th>
                     <th>Description</th>
                  </tr>
                 </thead>
                 <tbody>
                   <tr>
					 <td>Proportions heatmap</td>
					 <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/proportion_heatmap.R">https://github.com/CompEpigen/Decomp_web/blob/master/data/proportion_heatmap.R</a></td>
					 <td>An R script to generate the proportions heatmap of the TCGA-LUAD deconvolution experiment with 7 LMCs.</td>	
				  </tr>
 				  <tr>
					 <td>Trait association</td>
					 <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/trait_association.R">https://github.com/CompEpigen/Decomp_web/blob/master/data/trait_association.R</a></td>
					 <td>An R script to compare both quantitative and qualitative traits with LMC proportions.</td>
				  </tr>
				  <tr>
                     <td>Comparison with mutational data</td>
                     <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/compare_with_mutational_data.R">https://github.com/CompEpigen/Decomp_web/blob/master/data/compare_with_mutational_data.R</a></td>
                     <td>An R script to associate LMC proportions with various kinds of genetic alterations.</td>
                   </tr>
				  <tr>
                     <td>Comparison to cancer stemness</td>
                     <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/correlation_with_stemness.R">https://github.com/CompEpigen/Decomp_web/blob/master/data/correlation_with_stemness.R</a></td>
                     <td>An R script to associate LMC proportions with cancer stemness indices.</td>
                   </tr>
				  <tr>
                     <td>Differential LMC analysis</td>
                     <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/compare_LMCs_scatterplot.R">https://github.com/CompEpigen/Decomp_web/blob/master/data/compare_LMCs_scatterplot.R</a></td>
                     <td>An R script to compare LMCs to one another, for instance to create scatterplots describing multiple LMCs.</td>
                   </tr>
				  <tr>
                     <td>Proportions and gene expression</td>
                     <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/quantify_gene_expression.R">https://github.com/CompEpigen/Decomp_web/blob/master/data/quantify_gene_expression.R</a></td>
                     <td>An R script to compare LMC proportions with gene expression values of cell type marker genes in lung tissue.</td>
                   </tr>
				  <tr>
                     <td>Survival analysis</td>
                     <td><a href="https://github.com/CompEpigen/Decomp_web/blob/master/data/survival_analyis.R">https://github.com/CompEpigen/Decomp_web/blob/master/data/survival_analyis.R</a></td>
                     <td>An R script to perform survival analysis of patients with different LMC proportions.</td>
                   </tr>

                  </tbody>
               </table>
           </div>
        </div>
        <!-- /.row -->

    </div>
    <!-- /.container -->

    <!-- Footer -->
    <hr>
    <footer class="py-5 bg-inverse">
        <div class="container">           <p class="m-0 text-center text-white">Copyright &copy; <a href="mailto:michael.scherer@crg.eu">Support</a> 2021 </p>
        </div>
    </footer>

    <!-- jQuery -->
    <script src="js/jquery.js"></script>

    <!-- Bootstrap Core JavaScript -->
    <script src="js/bootstrap.min.js"></script>


</body>

</html>
